robot.py

```diff
@@ -107,49 +110,73 @@ def teleopInit(self):
 
     def teleopPeriodic(self):
         """Executed every cycle"""
-        self.handle_intake_inputs(self.driver_joystick)
-        self.handle_chassis_inputs(self.driver_joystick)
-        self.handle_spinner_inputs(self.driver_joystick)
-        self.handle_shooter_inputs(self.driver_joystick)
-        self.handle_hang_inputs(self.driver_joystick)
+        self.handle_intake_inputs(self.driver_joystick, self.codriver_gamepad)
+        self.handle_chassis_inputs(self.driver_joystick, self.codriver_gamepad)
+        self.handle_spinner_inputs(self.driver_joystick, self.codriver_gamepad)
+        self.handle_shooter_inputs(self.driver_joystick, self.codriver_gamepad)
+        self.handle_hang_inputs(self.driver_joystick, self.codriver_gamepad)
 
         self.shooter_controller.engage()
 
-    def handle_intake_inputs(self, joystick: wpilib.Joystick) -> None:
-        if joystick.getRawButtonPressed(6):
+    def handle_intake_inputs(
+        self, joystick: wpilib.Joystick, gamepad: wpilib.XboxController
+    ) -> None:
+        if joystick.getRawButtonPressed(2):
             if self.indexer.intaking:
                 self.indexer.disable_intaking()
+                self.indexer.raise_intake()
             else:
                 self.indexer.enable_intaking()
-
-    def handle_spinner_inputs(self, joystick: wpilib.Joystick) -> None:
+                self.indexer.lower_intake()
+
+        if gamepad.getAButtonPressed():
```

Comment:
I think you want getAButton() otherwise this will only run for 1 control loop

Reply:
Correct